src/sentry/static/sentry/app/views/settings/components/forms/richListField.jsx

```diff
@@ -0,0 +1,231 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import {t} from 'app/locale';
+import DropdownAutoComplete from 'app/components/dropdownAutoComplete';
+import DropdownButton from 'app/components/dropdownButton';
+import InputField from 'app/views/settings/components/forms/inputField';
+import InlineSvg from 'app/components/inlineSvg';
+import Confirm from 'app/components/confirm';
+
+const ItemList = styled('div')`
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const Item = styled('span')`
+  display: inline-block;
+  background-color: ${p => p.theme.button.default.background};
+  border: 1px solid ${p => p.theme.button.default.border};
+  border-radius: ${p => p.theme.button.borderRadius};
+  color: ${p => p.theme.button.default.color};
+  cursor: default;
+  font-size: ${p => p.theme.fontSizeSmall};
+  font-weight: 600;
+  line-height: 1;
+  padding: 0;
+  text-transform: none;
+  margin-right: 10px;
+  margin-bottom: 5px;
+`;
+
+const ItemLabel = styled('span')`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 8px 12px;
+  white-space: nowrap;
+`;
+
+const ItemIcon = styled('span')`
+  padding-left: 10px;
+  color: ${p => p.theme.gray2};
+  cursor: pointer;
+
+  &:hover {
+    color: ${p => p.theme.button.default.color};
+  }
+`;
+
+const RichListProps = {
+  /**
+   * Text used for the add item button.
+   */
+  addButtonText: PropTypes.node,
+
+  /**
+   * Configuration for the add item dropdown.
+   */
+  addDropdown: PropTypes.shape(DropdownAutoComplete.propTypes).isRequired,
+
+  /**
+   * Render function to render an item.
+   */
+  renderItem: PropTypes.func,
+
+  /**
+   * Callback invoked when an item is added via the dropdown menu.
+   */
+  onAddItem: PropTypes.func,
+
+  /**
+   * Callback invoked when an item is interacted with.
+   */
+  onEditItem: PropTypes.func,
+
+  /**
+   * Callback invoked when an item is removed.
+   */
+  onRemoveItem: PropTypes.func,
+
+  /**
+   * Properties for the confirm remove dialog. If missing, the item will be
+   * removed immediately.
+   */
+  removeConfirm: PropTypes.object,
+};
+
+function getDefinedProps(propTypes, props) {
+  return _.pickBy(props, (_prop, key) => key in propTypes);
+}
+
+class RichList extends React.PureComponent {
```

Comment:
👍 for the field name. We probably should stick `Field` as a suffix to be consistent (though I know the one I built didn't do this)

Reply:
This is not the field. This is the list within the field. The field is actually called `RichListField`.